### ISSUE TITLE
assert padding in {'same', 'valid', 'full'} AssertionError

### ISSUE THREAD

**Issue by thisismygitrepo** [tensorflow/tensorflow#41692]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below): tf-nightly
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

I'm using Conv1DTranspose and I set `padding` to `causal` which is only valid for Conv1D but it fails and gives the assertion above.

**Describe the expected behavior**
 Unless I'm very much mistaken this assertion should be updated.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
@kernelizd 

Request you to fill issue template.
Please, share colab link or simple standalone code to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by ymodak** [comment]
I tried with tf-nightly version 2.4.0-dev20200724 and it works. Can you please confirm?
```python
tf.keras.layers.Conv1DTranspose(filters = 2, kernel_size = 3, strides=1, padding='causal', output_padding=None)
#Outputs:
<tensorflow.python.keras.layers.convolutional.Conv1DTranspose at 0x7fc6bc89cf60>
```

**Comment by kernelizd** [comment]
@ymodak 

Thank you for trying it out. Yes, I can confirm the issue is there.
The model can be constructed, but onc you pass something you get the error.
I constructed the same layer as you did above, called it `m`, then I passed

`m(np.random.randn(4, 100, 2))`

And got the error in the title.

**Comment by ymodak** [comment]
As per the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv1DTranspose) the `causal` arg is not accepted as valid input for Conv1DTranspose.
